cmd/codegen/generator/go/templates/src/_dagger.gen.go/defs.go.tmpl

```diff
@@ -163,6 +165,13 @@ func getClientParams() (graphql.Client, *querybuilder.Selection) {
 	httpClient := &http.Client{
 		Transport: roundTripperFunc(func(r *http.Request) (*http.Response, error) {
 			r.SetBasicAuth(sessionToken, "")
+
+			// detect $TRACEPARENT set by 'dagger run'
+			r = r.WithContext(fallbackSpanContext(r.Context()))
```

Comment:
The `$TRACEPARENT` injection works only `dagger run` (or dagger call also, I'm not sure), right? Just thinking about auto-provision (automatic download the dagger binary), but that case looks impossible to set it.

Reply:
You're right, there's not much we can do when it's not wrapped in `dagger call` or `dagge run` since there's no opportunity to establish a root span - unless the user's code is doing that themselves. I think that's OK.

The comment says `dagger run` but this will also be the case when we're calling a module function, since the `shim` executable sets `$TRACEPARENT` there too. Will update the comment.